components/livekit/tests/livekit-component.spec.ts

```diff
@@ -0,0 +1,643 @@
+      listParticipants: jest.fn(),
+      removeParticipant: jest.fn(),
+      updateParticipant: jest.fn()
+    })),
+    IngressClient: jest.fn().mockImplementation(() => ({
+      listIngress: jest.fn(),
+      createIngress: jest.fn(),
+      deleteIngress: jest.fn()
+    })),
+    WebhookReceiver: jest.fn().mockImplementation(() => ({
+      receive: jest.fn()
+    })),
+    IngressInput: {
+      RTMP_INPUT: 0,
+      WHIP_INPUT: 1,
+      URL_INPUT: 2
+    },
+    TrackSource: {
+      CAMERA: 0,
+      MICROPHONE: 1,
+      SCREEN_SHARE: 2,
+      SCREEN_SHARE_AUDIO: 3
+    }
+  }
+})
+
+let logs: ILoggerComponent
+let component: ILivekitComponent
+let settings: LivekitSettings
+let mockRoomClient: any
+let mockIngressClient: any
+let mockWebhookReceiver: any
+
+beforeEach(async () => {
+  logs = createLoggerMockedComponent()
+  settings = {
+    host: 'wss://livekit.example.com',
+    apiKey: 'test-api-key',
+    secret: 'test-api-secret'
+  }
+
+  // Get the mock instances
+  const { RoomServiceClient, IngressClient, WebhookReceiver } = require('livekit-server-sdk')
+
+  component = await createLivekitComponent({ logs }, { settings })
+
+  // Get the mock instances that were created
+  mockRoomClient = RoomServiceClient.mock.results[0].value
+  mockIngressClient = IngressClient.mock.results[0].value
+  mockWebhookReceiver = WebhookReceiver.mock.results[0].value
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('when creating the livekit component', () => {
+  describe('and the host URL does not have wss:// prefix', () => {
+    it('should normalize the host URL by adding wss:// prefix', async () => {
+      const settingsWithoutProtocol = {
+        ...settings,
+        host: 'livekit.example.com'
+      }
+
+      const comp = await createLivekitComponent({ logs }, { settings: settingsWithoutProtocol })
+      const credentials = await comp.generateCredentials({
+        identity: 'test',
+        roomName: 'test-room'
+      })
+
+      expect(credentials.url).toBe('wss://livekit.example.com')
+    })
+  })
+
+  describe('and the host URL already has wss:// prefix', () => {
+    it('should keep the host URL as-is', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'test',
+        roomName: 'test-room'
+      })
+
+      expect(credentials.url).toBe('wss://livekit.example.com')
+    })
+  })
+})
+
+describe('when generating credentials', () => {
+  describe('and using default permissions', () => {
+    it('should generate credentials with url and token', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'user-123',
+        roomName: 'my-room'
+      })
+
+      expect(credentials).toEqual({
+        url: 'wss://livekit.example.com',
+        token: 'mock-jwt-token'
+      })
+    })
+  })
+
+  describe('and using custom permissions', () => {
+    it('should generate credentials with the provided permissions', async () => {
```

Comment:
We really can't know if we're generating the credentials with the provided permissions if we don't assert over the mock, WDYT about doing that?
The same seems to happen with other `generateCredentials` tests.